### PR TITLE
4 row v2-cards variant #593

### LINKED ISSUE #593: FR - MT Card Row Block (4 cards row variant)
From vg-macktrucks-com-rd created by [DanielaPedrochevd](https://github.com/DanielaPedrochevd): Netcentric/vg-macktrucks-com-rd#187  

**Agile Requirement**
As a business owner, I want to display content so that it follows the design, to achieve a more dynamic site: 

- [4 Cards Row](https://www.figma.com/file/URJhT8pkntqc6h6v5wSLqx/Mack-Core-Design-Kit?type=design&node-id=478-32530&mode=dev)

**Description**

Create a new Card Row Block that contains a variant (4 Card Row block) that follows the above-mentioned design. 

**Acceptance Criteria**
1. As an editor, I can add a new "Card Row" block to any page. 
2. As an editor, I can add images and links to these cards so that users are redirected to them upon click. 
3. As an editor, I can select the 4 cards row variant (default variant).
4. For tablet viewport, need a 2-column grid with 2 cards per row. 
5. Editors responsible for selecting the correct variant.

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
Fix #593

Test URLs:
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/jlledo/v2/v2-cards-demo
- After: https://593-4-cards-row-variant--vg-macktrucks-com--hlxsites.hlx.page/drafts/jlledo/v2/v2-cards-demo

### CODE REVIEW

File: blocks/v2-cards/v2-cards.css [hunk 1]
```diff
@@ -85,6 +85,42 @@
   height: 16px;
 }
 
+/* 4 rows variant */
```
Comment (amglebihan) [inline blocks/v2-cards/v2-cards.css:88]: The name should be "4 cards row".

File: blocks/v2-cards/v2-cards.css [hunk 2]
```diff
@@ -85,6 +85,42 @@
...
+.section:has(.v2-cards--4-rows) {
+  background-color: var(--c-primary-gray);
```
Comment (amglebihan) [inline blocks/v2-cards/v2-cards.css:90]: I believe we should remove that option and allow the content editors to choose the background colour of the section, as it is customisable.

File: blocks/v2-cards/v2-cards.css (hunk 1 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-cards/v2-cards.css]: Done

File: blocks/v2-cards/v2-cards.css (hunk 2 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-cards/v2-cards.css:90]: added as new section metadata backgound variant

File: styles/styles.css [hunk 3]
```diff
@@ -904,6 +904,7 @@ main .section.responsive-title h1 {
   --testimonial-letter-spacing: -0.0225rem;
 
   /* In page navigation */
+  /* stylelint-disable-next-line length-zero-no-unit */
```
Comment (amglebihan) [inline styles/styles.css:907]: You can just remove "px" in the line below instead of adding that comment.

File: blocks/v2-cards/v2-cards.css [hunk 4]
```diff
@@ -85,6 +85,38 @@
...
+  justify-content: space-between;
+  align-items: center;
+  padding: var(--card-container-padding);
+  font-size: 18px;
+}
+
+.v2-cards--4-cards-row .v2-cards__button-container .icon,
+.v2-cards--4-cards-row .v2-cards__button-container svg {
+  width: 22px;
+  height: 22px;
```
Comment (santi-homps) [inline blocks/v2-cards/v2-cards.css:113]: shouldn't this values be 24px for consistency?

File: blocks/v2-cards/v2-cards.css [hunk 5]
```diff
@@ -85,6 +85,38 @@
...
+.v2-cards--4-cards-row .v2-cards__button-container {
+  margin-top: 0;
+}
+
+.v2-cards--4-cards-row .v2-cards__button-container .v2-cards__button:any-link {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: var(--card-container-padding);
+  font-size: 18px;
```
Comment (santi-homps) [inline blocks/v2-cards/v2-cards.css:107]: `--body-font-size-m` is not 18px?

**Review by santi-homps — APPROVED** [review]

File: blocks/v2-cards/v2-cards.css (hunk 5 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-cards/v2-cards.css:107]: it is. good catch, thanks. Done

File: blocks/v2-cards/v2-cards.css [hunk 6]
```diff
@@ -115,4 +147,20 @@
...
     margin: auto;
   }
+
+  /* 4 cards row variant */
+  .v2-cards--4-cards-row {
+    --card-container-padding: 16px;
+  }
+
+  .v2-cards--4-cards-row .v2-cards__card-item {
+    max-width: calc((100% - 32px) / 2);
```
Comment (alexiscoelho) [inline blocks/v2-cards/v2-cards.css:157]: Instead of `32px` -> `(2 x var(--card-container-padding))`

File: blocks/v2-cards/v2-cards.css [hunk 7]
```diff
@@ -115,4 +147,20 @@
...
+  }
+}
+
+@media (min-width: 1200px) {
+  /* 4 cards row variant */
+  .v2-cards--4-cards-row .v2-cards__card-item {
+    max-width: calc((100% - 48px) / 4);
```
Comment (alexiscoelho) [inline blocks/v2-cards/v2-cards.css:164]: Same here for the `48px`

**Review by alexiscoelho — APPROVED** [review]

File: blocks/v2-cards/v2-cards.css (hunk 7 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-cards/v2-cards.css:164]: done

**Pull request merged**